### PR TITLE
Enable updating additional account information from user preferences via rest api

### LINKED ISSUE #6553: Allow syncing of privacy/sensitive preferences across apps
This is an often requested feature for Tusky. It is already possible to retrieve the preferences via `GET /api/v1/accounts/verify_credentials`, but not to set them via the api. It would be super awesome if that feature could be added. Unfortunately I don't understand enough of Ruby to send a pull request myself.

* * * *

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [ ] This bug happens on a [tagged release](https://github.com/tootsuite/mastodon/releases) and not on `master` (If you're a user, don't worry about this).

### PR DESCRIPTION
**Pull request by Tak**
Resolves #6553

### CODE REVIEW

File: app/controllers/api/v1/accounts/credentials_controller.rb
Comment (akihikodaki) [inline app/controllers/api/v1/accounts/credentials_controller.rb]: Why renaming `setting_default_privacy` as `privacy`? Indeed `setting` is kind of duplicate and fine to remove, but `default` is not such a duplicate and should be preserved.

File: app/controllers/api/v1/accounts/credentials_controller.rb
Comment (akihikodaki) [inline app/controllers/api/v1/accounts/credentials_controller.rb]: Why not create a hash at first place if you eventually converts to hash?

File: app/controllers/api/v1/accounts/credentials_controller.rb
Comment (akihikodaki) [inline app/controllers/api/v1/accounts/credentials_controller.rb]: It's not database who doesn't like boolean value, but `UserSettingsDecorator` is. `UserSettingsDecorator` should be updated instead of adding a workaround.

File: spec/controllers/api/v1/accounts/credentials_controller_spec.rb
Comment (akihikodaki) [inline spec/controllers/api/v1/accounts/credentials_controller_spec.rb]: Thanks for spec. I appreciate it.

**Review by akihikodaki — CHANGES_REQUESTED** [review]

File: app/controllers/api/v1/accounts/credentials_controller.rb
Comment (Tak) [reply app/controllers/api/v1/accounts/credentials_controller.rb]: This is because that's the way it goes out in `verify_credentials` - the user of the api just modifies the values in the structure she receives and sends it back.

File: app/controllers/api/v1/accounts/credentials_controller.rb
Comment (Tak) [reply app/controllers/api/v1/accounts/credentials_controller.rb]: Ok, I can do that. I wasn't sure about the additional effects of making a change like that.

File: app/controllers/api/v1/accounts/credentials_controller.rb
Comment (Tak) [reply app/controllers/api/v1/accounts/credentials_controller.rb]: I tried just returning a hash that seemed identical to me, but the update failed when I did that, and I'm not sure why. I assumed there was some hidden side effect of doing it this way...

Comment (Tak) [reply app/controllers/api/v1/accounts/credentials_controller.rb]: ...and there is - the key symbols are converted to strings. Updating...

File: app/controllers/api/v1/accounts/credentials_controller.rb
Comment (akihikodaki) [inline app/controllers/api/v1/accounts/credentials_controller.rb]: You don't need `permit` here.

File: app/controllers/api/v1/accounts/credentials_controller.rb
Comment (akihikodaki) [inline app/controllers/api/v1/accounts/credentials_controller.rb]: `fetch` method should be convenient for the purpose.

File: app/lib/user_settings_decorator.rb
Comment (akihikodaki) [inline app/lib/user_settings_decorator.rb]: Instead of adding decoercing operation, let's use `ActiveModel::Type::Boolean.new.cast` in `coerce_values`. It accepts both of boolean and numbers.

**Review by akihikodaki — CHANGES_REQUESTED** [review]

**Comment by Tak** [comment]
Is there any further action I need to take on this? Or it's just waiting for final review?

**Review by akihikodaki — APPROVED** [review]

**Pull request merged**
